tensorflow/core/kernels/mkl_slice_op.cc

```diff
@@ -397,10 +398,14 @@ class MklSliceOp : public OpKernel {
         size_dims = MklDnnDimsInNCHW(size_dims, input_tf_format);
         auto input_md = input_mkl_shape.GetMklLayout();
         src.SetUsrMem(input_md, &input_tensor);
+
+        // Handle data format safely, change them to block format.
+        // Compute parameters of reorder primitive first.
+        input_dims = input_mkl_shape.GetSizesAsMklDnnDims();
+        input_strides = CalculateTFStrides(input_dims);
```

Comment:
`input_strides` calculation can be refactored out if the `if`-`else`.

Reply:
I don't quite know how to move out the `input_strides`. 
1. If we move `input_strides` forward, before `if`: obviously this can not be done, because it relies on `input_dims` which is decided in `if`-`else`;
2. Otherwise if we move 'input_strides' backward, after 'else': also can not be done, since in the `else` clause `input_strides` is required to calculate `input_md`.

Any suggestions?

Reply:
My bad! The PR only shows two lines before and after each change. The two lines calculating `input_md` happen to be hidden and I overlooked. This is fine then. Sorry for the inconvenience!